### ISSUE TITLE
uutils/coreutils

### ISSUE THREAD

**Issue by NeQuissimus** [NixOS/nixpkgs#28114]
I would absolutely love for somebody to package up https://github.com/uutils/coreutils

:)

**Comment by Ma27** [comment]
will attempt to reserve some time to give it a try in the next days :-)

**Comment by Ma27** [comment]
@NeQuissimus I created a first working version which compiles on my machine ( `17.09pre112899.129f8d7e99 (Hummingbird)`): https://gist.github.com/Ma27/4390346ededcb41ddd9347b620ec2490

Unfortunately I had to use an older revision since the latest revisions require language features from Rust 1.19 and I'll have to bump it in nixpkgs before using the latest revision, after that I'll open a PR :-)

**Comment by Ma27** [comment]
btw do we want to add an option which allows to change the coreutils pkg from `coreutils` to `uutils-coreutils`?
Something like `environment.coreUtils = pkgs.coreutils;` and every module should start start using `cfg.environment.coreUtils`.

**Comment by NeQuissimus** [comment]
I had contemplated using a package override and just make `coreutils` equal to `uutils-coreutils`. Not sure how well that would work out...
This may be something where @edolstra is one of the few qualified to chime in.

**Comment by fpletz** [comment]
Please don't introduce this as a NixOS option. This rewrite is not compatible with the GNU coreutils. This will certainly break NixOS. The package override will break lots of packages and will trigger a full rebuild anyway.

Packaging these for testing is fine. :+1:

**Comment by NeQuissimus** [comment]
@Ma27 Could you just wrap this into a PR then? I will certainly test this and create some aliases to replace the GNU tools (in the user environment)

**Comment by Ma27** [comment]
sorry for the delay at first.

> This rewrite is not compatible with the GNU coreutils. 

I'm aware of this. I just proposed an approach to switch the `coreutils` package just for system-wide testing purposes.

> I had contemplated using a package override and just make coreutils equal to uutils-coreutils

This is actually the better approach. Furthermore this is something anyone can do locally, so @fpletz shouldn't have concers with it as it shouldn't affect code in `nixpkgs` for now :-)

> Could you just wrap this into a PR then?

I'll certainly do. However we have to wait for #28459 since the latest revisions of `uutils-coreutils` require Rust 1.19 (I'm currenty using an older revision from July 2017 locally). I checked out this change tried to compile uutils-coreutils against the new Rust compiler, but it cargo builds seem to fail (https://github.com/NixOS/nixpkgs/pull/28459#issuecomment-324858946)
